pkg/controller/directvolumemigration/rsync.go

```diff
@@ -226,7 +227,11 @@ func (t *Task) createRsyncConfig() error {
 	for ns, vols := range pvcMap {
 		pvcList := []pvc{}
 		for _, vol := range vols {
-			pvcList = append(pvcList, pvc{Name: vol.Name})
+			dnsSafeName, err := getDNSSafeName(vol.Name)
```

Comment:
Are there different rules for naming Pods and Volumes?

Reply:
yeah sorry I buried the link in the BZ, will put it in the PR description: https://github.com/kubernetes/kubernetes/blob/9cdd673a8b784f22709b133dde1da16332b14889/staging/src/k8s.io/api/core/v1/types.go#L39, it seems like the volume name rule is special

Reply:
**Most resource types require a name that can be used as a DNS subdomain name as defined in RFC 1123. This means the name must:**
- contain no more than 253 characters
- contain only lowercase alphanumeric characters, '-' or '.'
- start with an alphanumeric character
- end with an alphanumeric character

**Some resource types require their names to follow the DNS label standard as defined in RFC 1123. This means the name must:**
- contain at most 63 characters
- contain only lowercase alphanumeric characters or '-'
- start with an alphanumeric character
- end with an alphanumeric character